src/adaptations/device-layer/LwIP/WarmSupport.cpp

```diff
@@ -358,27 +383,66 @@ PlatformResult AddRemoveThreadAddress(InterfaceType inInterfaceType, const Inet:
 
 #if WARM_CONFIG_SUPPORT_THREAD_ROUTING
 
-#error "Weave Thread router support not implemented"
-
 PlatformResult StartStopThreadAdvertisement(InterfaceType inInterfaceType, const Inet::IPPrefix &inPrefix, bool inStart)
 {
-    // TODO: implement me
+    if (inInterfaceType != kInterfaceTypeThread) { return kPlatformResultFailure; }
```

Comment:
The codestyle is not consistent with others in weave, please address it.
Please use one error return statement for function.
please use different error for both if.